Backpack/src/main/java/net/skyscanner/backpack/chip/internal/BpkChipAppearances.kt

```diff
@@ -56,12 +56,14 @@ internal sealed class BpkChipAppearances : BpkChipAppearance {
         disabled = context.getColor(R.color.bpkTextDisabled),
       )
 
-    internal class Default(context: Context, typedArray: TypedArray?) : Solid(context) {
+    internal class Default(context: Context, private val typedArray: TypedArray?) : Solid(context) {
       constructor(
         context: Context,
         attrs: AttributeSet?,
         defStyleAttr: Int,
-      ) : this(context, context.theme.obtainStyledAttributes(attrs, R.styleable.BpkChip, defStyleAttr, 0))
+      ) : this(context, context.theme.obtainStyledAttributes(attrs, R.styleable.BpkChip, defStyleAttr, 0)) {
+        typedArray?.recycle()
```

Comment:
recycling it here where we crate it, rather than within the function. otherwise the passed in typed array might be recycled twice